platforms.go

```diff
@@ -283,7 +282,8 @@ func Format(platform specs.Platform) string {
 // For example, if "Aarch64" is encountered, we change it to "arm64" or if
 // "x86_64" is encountered, it becomes "amd64".
 func Normalize(platform specs.Platform) specs.Platform {
-	platform.OS = normalizeOS(platform.OS)
+	OSAndVersion := fmt.Sprintf(OSAndVersionFormat, platform.OS, platform.OSVersion)
```

Comment:
Formatting a string and then passing it to a function that parses that string is not the best pattern. I think better approach would be to split `normalizeOS` into two functions where one takes os and version arguments separately, and the second parses the string and then calls the first function.

Reply:
@tonistiigi that is what I had originally and I have review feedback to combine them both to one.

Reply:
@mikebrow

Reply:
in the prior .. by memory the part[0] already had both os and version in one string..

which was being passed to both the normalize for OS func and then also to the normalize for OS version func... at that point my recommendation was to not parse the same string twice with two normalize funcs one after the other..

Reply:
Given two patterns of use here.. one where the OS and version are in a string that need to be split and another where they are already split.. Yes, we should have a first func that takes the string performs the split then calls the common normalization code from the split vars..

Should also take a look at thaJeztah's proposal he's found possible errors in the reg use.

Reply:
here in this path the public Normalize already has the pieces separate.. 

So yes, I agree with tonistiligi, the "better approach would be to split normalizeOS into two functions where one takes os and version arguments separately, and the second parses the string and then calls the first function.  

My issue was with having two normalize funcs one for os and another for version.. not one for split and another with the two arguments already split.